src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,261 @@ public static class NUnitTestExtractorApp
         {
-            if (!string.IsNullOrEmpty(includeInfo))
-            {
-                //Remove quotes from string in order to properly process it
-                includeInfo = includeInfo.Replace("\"", "");
+            HashSet<Type> testSuites = new HashSet<Type>();
+
+            LoadAssembly(assemblyFullPath);
 
-                List<string> includeInfoValues = new List<string>(includeInfo.Split(','));
-                
-                foreach (string value in includeInfoValues)
+            foreach (var item in _assembly.GetTypes())
+            {
+                if (IsValidTestSuite(item))
                 {
-                    if (categoryNames != null && !categoryNames.Contains(value))
-                    {
-                        return false;
-                    }
+                    testSuites.Add(item);
                 }
-                return true;
             }
-            throw new ArgumentNullException("includeInfo", "no includeInfo specified");
 
+            return testSuites;
         }
 
         /// <summary>
-        /// Used to determine whether or not a test/testcase should be excluded or not
+        /// Get all valid tests from the suite
         /// </summary>
-        /// <param name="categoryNames">List of categories to compare info to</param>
-        /// <param name="excludeInfo">string of info used to compare to categoryNames</param>
-        /// <returns>returns true if excludeInfo is inside of the categoryNames List, false otherwise</returns>
-        public static bool Exclude(IReadOnlyList<string> categoryNames, string excludeInfo)
+        /// <param name="testSuite">A valid NUnit test suite</param>
+        /// <returns>Return a list of tests</returns>
+        private static HashSet<string> GetValidTestsFromTestSuite(Type testSuite)
         {
-            if (!string.IsNullOrEmpty(excludeInfo))
+            HashSet<string> tests = new HashSet<string>();
+
+            foreach (MethodInfo test in testSuite.GetMethods())
             {
-                excludeInfo = excludeInfo.Replace("\"", "");
+                var attributes = test.GetCustomAttributesData();
 
-                List<string> excludeInfoValues = new List<string>(excludeInfo.Split(','));
+                // if the function attributes don't have either"Test" or "TestCase" attribute, it is not a valid NUnit test, skip it.
+                if (!attributes.Any(x => x.AttributeType.Equals(typeof(TestAttribute)) || x.AttributeType.Equals(typeof(TestCaseAttribute))))
+                {
+                    continue;
+                }
 
-                foreach (string value in excludeInfoValues)
+                // if the function attributes have either "Explicit" or "Ignore" attribute, it is not a valid NUnit test, skip it.
+                if (attributes.Any(x => x.AttributeType.Equals(typeof(ExplicitAttribute))) || attributes.Any(x => x.AttributeType.Equals(typeof(IgnoreAttribute))))
```

Comment:
[Should] When you merge my change into your branch, you can remove these lines and call IsValidTestMethod() instead.